addons/website/models/website.py

```diff
@@ -927,16 +936,26 @@ def get_current_website(self, fallback=True):
         if website_id:
             return self.browse(website_id)
 
-        if not request and not fallback:
+        if not is_frontend_request and not fallback:
```

Comment:
`and` vs `OR` ?

is_frontend, not fallback -> should enter here imo... no ?

Reply:
I don't think so, the `fallback` is just about ensuring to return a record if we did not find a website matching:
- the session
- or the context
- or the request domain

If it is a `frontend` request, it is by definition a website context -> `get_current_website()` should return the current website IMO.

Or do you have an use case in mind?

Anyway, I don't think such a case will end up here as technically if we are in a `frontend` request, a website will be bound to the request (and so to the context as both are set at the same time) so it will always return before reaching this.

What do you think?

Reply:
case should probably never happen's but in case you are in a frontend request (for a louch reason without context propaged or force_website and with wrong domain) ... If I call explicitely with fallback=False, I'm waiting that it will respect the fallback param.

So yes it should not happen, but if it happens why not make the logic correct and respect this parameter ? It will probably never happens but it will be correct :innocent: 

Here in case of frontend request, you just ignore the param.

BTW, not critical to merge since it is the worst case. The bug that you fix with this one is more important than this nitpicking on the code ;) Thanks your the fix !

Reply:
> If I call explicitely with fallback=False, I'm waiting that it will respect the fallback param.

It will: the `fallback` param is about returning or not a random website if the request domain does not match any website, it's not about doing the domain matching search or not, right?
So in your case, you will go through the domain match, and if it does not find a website matching it, it will not return a random website if you passed `fallback=False`, respecting it.

Reply:
:+1: it will be respected here https://github.com/odoo/odoo/pull/118526/files#diff-5e92e473fa4d3da6db7ef727fb217dad51ef6c2383913edca73fe040a23e82c2R1009